src/cpp/astar.cpp

```diff
@@ -36,6 +38,44 @@ inline float l1_norm(int i0, int j0, int i1, int j1) {
   return std::abs(i0 - i1) + std::abs(j0 - j1);
 }
 
+// Please note below heuristic is experimental and only for pretty lines
+// Orthogonal x (moves by x first, then half way by y)
+inline float orthogonal_x(int i0, int j0, int i1, int j1, int i2, int j2) {
+  int di = std::abs(i0 - i1);
+  int dim = std::abs(i1 - i2);
+  int djm = std::abs(j1 - j2);
+  if (di > (dim * 0.5)) {
+    return di + djm;
+  } else {
+    return std::abs(j0 - j1);
+  }
+}
+
+// Please note below heuristic is experimental and only for pretty lines
+// Orthogonal y (moves by y first, then half way by x)
+inline float orthogonal_y(int i0, int j0, int i1, int j1, int i2, int j2) {
+  int dj = std::abs(j0 - j1);
+  int djm = std::abs(j1 - j2);
+  int dim = std::abs(i1 - i2);
+  if (dj > (djm * 0.5)) {
+    return dj + dim;
+  } else {
+    return std::abs(i0 - i1);
+  }
+}
+
+typedef float (*heuristic_ptr)(int, int, int, int, int, int);
+
+inline heuristic_ptr select_heuristic(int h) {
```

Comment:
I don't see a good reason for this function to be `inline`. It's only called once after all. Remove `inline`?

Reply:
yeah, given we using a pointer it probably doesn't make sense anyway.